test/functional/autocmd/textyankpost_spec.lua

```diff
@@ -0,0 +1,166 @@
+local helpers = require('test.functional.helpers')
+local clear, eval, eq, insert = helpers.clear, helpers.eval, helpers.eq, helpers.insert
+local feed, execute, expect = helpers.feed, helpers.execute, helpers.expect
+
+
+describe('TextYankPost', function()
+  before_each(function()
+    clear()
+
+    -- emulate the clipboard so system clipboard isn't affected
+    execute('let &rtp = "test/functional/fixtures,".&rtp')
+
+    execute('let g:count = 0')
+    execute('autocmd TextYankPost * let g:event = copy(v:event)')
+    execute('autocmd TextYankPost * let g:count += 1')
```

Comment:
What does the pattern match against? I think in the doc, we should mention that the pattern should always be `*`, and that it may change in the future to match against something more useful than "the current filename".